### PR TITLE
Disable traps when z10 architecture is disabled

### PR DESCRIPTION
**Pull request by NigelYiboYu**
If one chooses to disable z10 architecture level, traps should be disabled
so that the optimizations and codegen can avoid generating trap-related IL
nodes and instructions.

Move project specific random z architecture level disabling code out of
OMR and into the relevant project.

### CODE REVIEW

File: compiler/z/codegen/OMRCodeGenerator.cpp [hunk 1]
```diff
@@ -552,65 +552,16 @@ OMR::Z::CodeGenerator::CodeGenerator()
...
+   
    // Check for -Xjit disable options and target this specific compilation for the proper target
    if (comp->getOption(TR_DisableZ10))
       {
       _processorInfo.disableArch(TR_S390ProcessorInfo::TR_z10);
+
+      // If z10 is disabled, there will no longer be trap instructions.
+      // Set disable traps so that the optimizations and codegen can avoid generating
+      // trap-specific nodes or instructions.
+      comp->setOption(TR_DisableTraps);
```
Comment (fjeremic) [inline compiler/z/codegen/OMRCodeGenerator.cpp]: This code doesn't belong here. What if the actual hardware we're running on is z9 and the user didn't specify any compiler options? You would never enter this path and `comp->getOption(TR_DisableTraps)` would return `true`. 

Please move this code to other z10 related queries below:
https://github.com/eclipse/omr/blob/d183d21dc7304ed1f39039bed26c81be9cc5ff2e/compiler/z/codegen/OMRCodeGenerator.cpp#L753-L756

File: compiler/z/codegen/OMRCodeGenerator.hpp [hunk 2]
```diff
@@ -1178,6 +1173,12 @@ class OMR_EXTENSIBLE CodeGenerator : public OMR::CodeGenerator
    TR::list<TR::S390ConstantDataSnippet*>  _constantList;
    TR::list<TR::S390ConstantDataSnippet*>  _snippetDataList;
 
+   /**
+    * _processorInfo contains the targetted hardware level for the compilation.
+    * This may be different than the real hardware the JVM is currently running
+    * on, due to user specified options.
+    */
+   TR_S390ProcessorInfo            _processorInfo;
```
Comment (fjeremic) [inline compiler/z/codegen/OMRCodeGenerator.hpp]: Awesome to see people documenting related code as part of their changes. Keep it up!

File: compiler/z/codegen/OMRCodeGenerator.cpp (hunk 1 above)
Comment (NigelYiboYu) [inline compiler/z/codegen/OMRCodeGenerator.cpp]: OK. Didn't realize that we didn't have existing code to disable traps on z9.

File: compiler/z/codegen/OMRCodeGenerator.hpp [hunk 3]
```diff
@@ -1178,6 +1173,12 @@ class OMR_EXTENSIBLE CodeGenerator : public OMR::CodeGenerator
    TR::list<TR::S390ConstantDataSnippet*>  _constantList;
    TR::list<TR::S390ConstantDataSnippet*>  _snippetDataList;
 
+   /**
+    * _processorInfo contains the targetted hardware level for the compilation.
+    * This may be different than the real hardware the JVM is currently running
```
Comment (0xdaryl) [inline compiler/z/codegen/OMRCodeGenerator.hpp]: Is this Java specific?  Or should "JVM" be rephrased to something runtime-neutral?

File: compiler/z/codegen/OMRCodeGenerator.hpp [hunk 4]
```diff
@@ -1178,6 +1173,12 @@ class OMR_EXTENSIBLE CodeGenerator : public OMR::CodeGenerator
    TR::list<TR::S390ConstantDataSnippet*>  _constantList;
    TR::list<TR::S390ConstantDataSnippet*>  _snippetDataList;
 
+   /**
+    * _processorInfo contains the targetted hardware level for the compilation.
```
Comment (0xdaryl) [inline compiler/z/codegen/OMRCodeGenerator.hpp]: Spelling: targeted

**Comment by NigelYiboYu** [comment]
1. Fixed typo in comments and also replaced the word `JVM` with something project neutral.
2. Split into two commits for logical separation

**Comment by 0xdaryl** [comment]
Thanks @NigelYiboYu .  Please remove the "Task:" from your commit messages as it is meaningless to Eclipse OMR.  After that, this should be good to merge.

**Comment by NigelYiboYu** [comment]
@0xdaryl `Task` numbers removed.

**Pull request merged**